src/bootstrap/bin/bootstrap-shim.rs

```diff
@@ -0,0 +1,75 @@
+use std::{
+    env, io,
+    process::{self, Command, ExitStatus},
+};
+
+use bootstrap::{Flags, MinimalConfig};
+
+#[path = "../../../src/tools/x/src/main.rs"]
+mod x;
+
+fn find_and_run_py_bootstrap_script() {
+    #[cfg(unix)]
+    fn exec_or_status(command: &mut Command) -> io::Result<ExitStatus> {
+        use std::os::unix::process::CommandExt;
+        Err(command.exec())
+    }
+
+    #[cfg(not(unix))]
+    fn exec_or_status(command: &mut Command) -> io::Result<ExitStatus> {
+        command.status()
+    }
+
+    let current_path = match env::current_dir() {
+        Ok(dir) => dir,
+        Err(err) => {
+            eprintln!("Failed to get current directory: {err}");
+            process::exit(1);
+        }
+    };
+
+    for dir in current_path.ancestors() {
+        let candidate = dir.join("x.py");
+        if candidate.exists() {
+            let mut cmd: Command;
+            cmd = Command::new(x::python());
+            cmd.arg(&candidate).args(env::args().skip(1)).current_dir(dir);
+            let result = exec_or_status(&mut cmd);
+
+            match result {
+                Err(error) => {
+                    eprintln!("Failed to invoke `{:?}`: {}", cmd, error);
+                }
+                Ok(status) => {
+                    process::exit(status.code().unwrap_or(1));
+                }
+            }
+        }
```

Comment:
This code looks near-identical to the x tool's find_and_run_available_bootstrap_script -- why do we limit ourselves to py here? Can we add a comment on this function and that function explaining the duplication?